### PR TITLE
feat: optimize write and read for snowflake

### PR DESCRIPTION
**Pull request by chamini2**
https://github.com/snowflakedb/snowflake-connector-python/issues/544 manually resolved

Merge #487 before and use it's introduced artifact creation for tests.

<details>
Affected

- bigquery
- snowflake
</details>

### CODE REVIEW

File: tests/_fal_testing/utils.py [hunk 1]
```diff
@@ -1,29 +1,31 @@
...
+    return output
 
 
-def create_script_artifact(context, prompt):
-    create_artifact(context, f".{prompt}.txt")
+def create_model_artifact(context, **kwargs):
+    create_artifact(context, ".txt", **kwargs)
 
 
-create_before_script_artifact = partial(create_script_artifact, prompt="before")
```
Comment (isidentical) [inline tests/_fal_testing/utils.py:26]: Why are we removing these? (I also decided to remove them in the structured hooks PR, but not sure why their removal is needed here)

Comment (chamini2) [reply tests/_fal_testing/utils.py:26]: not needed, more of an opinion. It came from needing to replace the content created inside the artifact.

Comment (isidentical) [reply tests/_fal_testing/utils.py:26]: Wdyt about the approach in the structed hooks PR (`create_dynamic_artifact`)?

Comment (chamini2) [reply tests/_fal_testing/utils.py:26]: I think that works. I can remove these changes.

Comment (isidentical) [reply tests/_fal_testing/utils.py:26]: Thanks!

File: src/faldbt/lib.py [hunk 2]
```diff
@@ -688,3 +688,87 @@ def _bigquery_write_relation(
...
+                schema=schema,
+                auto_create_table=True,
+                quote_identifiers=False,
+            )
+            if not success:
+                # In case the failure does not raise by itself
+                # I have not been able to reproduce such a case
+                from dbt.exceptions import DatabaseException
+
+                raise DatabaseException(output)
```
Comment (isidentical) [inline src/faldbt/lib.py:771]: Nice!

**Review by isidentical — APPROVED** [review]

**Pull request merged**